MBBSEmu.Tests/ExportedModules/Majorbbs/file_Tests.cs

```diff
@@ -300,5 +300,329 @@ public void fclose_invalidPointer()
+
+            var filep = fopen("FILE.TXT", "r");
+            Assert.NotEqual(0, filep.Segment);
+            Assert.NotEqual(0, filep.Offset);
+
+            Assert.Equal(0, fseek(filep, fseekOffset, 0));
+
+            Assert.Equal((byte)getValue, fgetc(filep));
+
+            Assert.Equal((byte)ungetValue, ungetc(ungetValue, filep));
+
+            var curFileStruct = new FileStruct(mbbsEmuMemoryCore.GetArray(filep, FileStruct.Size));
+            var curFileStream = majorbbs.FilePointerDictionary[curFileStruct.curp.Offset];
+            Assert.Equal(ungetValue, curFileStream.ReadByte());
+
+            Assert.Equal(0, fclose(filep));
+        }
+
+        [Theory]
+        [InlineData("Mary Had a Little Lamb\0", 0)]
+        [InlineData("---+++---111\0", 10)]
+        [InlineData("((((jumbo)))))\0", 20)]
+        [InlineData(" yeah \0", 25)]
+        [InlineData("\0", 40)]
+        [InlineData("A\0", 0)]
+        public void fputs_fgets_fseek_origin0_file(string stringPut, int fseekOffset)
+        {
+            //Reset State
+            Reset();
+
+            var filePath = CreateTextFile("file.txt", LOREM_IPSUM);
+            Assert.Equal(LOREM_IPSUM.Length, new FileInfo(filePath).Length);
+
+            var stringPutPtr = mbbsEmuMemoryCore.AllocateVariable("STRING_PUT", (ushort)stringPut.Length);
+            mbbsEmuMemoryCore.SetArray(stringPutPtr, Encoding.ASCII.GetBytes(stringPut));
+
+            var stringGetPtr = mbbsEmuMemoryCore.AllocateVariable("STRING_GET", (ushort)stringPut.Length);
+
+            var filep = fopen("FILE.TXT", "w");
+            Assert.NotEqual(0, filep.Segment);
+            Assert.NotEqual(0, filep.Offset);
+
+            Assert.Equal(0, fseek(filep, fseekOffset, 0));
+
+            Assert.Equal(1, fputs(stringPutPtr, filep));
+
+            Assert.Equal(0, fseek(filep, fseekOffset, 0));
+
+            Assert.Equal(stringGetPtr, fgets(stringGetPtr, (ushort)stringPut.Length, filep));
+            Assert.Equal(stringPut, Encoding.ASCII.GetString(mbbsEmuMemoryCore.GetString("STRING_GET")));
+
+            Assert.Equal(0, fclose(filep));
+        }
+
+        [Theory]
+        [InlineData(4, 'm')]
+        [InlineData(10, 'm')]
+        [InlineData(20, 't')]
+        [InlineData(25, 't')]
+        [InlineData(55, ',')]
+        public void fgetc_fseek_origin1_file(int fseekOffset, byte expectedChar)
+        {
+            //Reset State
+            Reset();
+
+            var filePath = CreateTextFile("file.txt", LOREM_IPSUM);
+
+            Assert.Equal(LOREM_IPSUM.Length, new FileInfo(filePath).Length);
+
+            var filep = fopen("FILE.TXT", "r");
+            Assert.NotEqual(0, filep.Segment);
+            Assert.NotEqual(0, filep.Offset);
+
+            Assert.Equal(0, fseek(filep, 0, 0));
+
+            Assert.Equal(0, fseek(filep, fseekOffset, 1));
+
+            Assert.Equal(expectedChar, fgetc(filep));
+
+            Assert.Equal(0, fclose(filep));
+        }
+
+        [Theory]
```

Comment:
could probably combine this with the test above it adding a third parameter to be passed to `fseek`